### DISCUSSION TITLE
[Proposal] About null-coalescening with throws (with discards)

### DISCUSSION THREAD

**Discussion by Alysaar** [dotnet/csharplang#4179] [Language Ideas] [answered]
I see a lot of people suggesting `_ = arg ?? throw new ArgumentNullException(nameof(arg));` instead of good old ifs to check for null arguments.

While I certainly see the appeal of this shorter syntax, it feels like a perversion of how discards are supposed to be used.
Indeed, the discard is here purely for syntax purposes.

I suggest allowing throws on the right-hand side without assignments, like so :
`arg ?? throw new ArgumentNullException(nameof(arg));`

**Comment by huoyaoyuan** [comment, marked as answer]
#2145 is covering the main scenarios
